drivers/flash/soc_flash_nrf.c

```diff
@@ -142,6 +147,12 @@ static int flash_nrf_read(struct device *dev, off_t addr,
 		return 0;
 	}
 
+#if CONFIG_ARM_NONSECURE_FIRMWARE
+	if (addr < PM_APP_ADDRESS) {
```

Comment:
I see silent assumption that secure flash partition to read always begin at 0 and ends up at PM_APP_ADDRESS

Reply:
That is a sane assumption, also made in the SPM and secure services.

In theory it is not always like this, in practice (for these solutions) it will always be contiguous.

Reply:
Ok, can live with this.

Reply:
Maybe add a comment which documents this so that customers who looks to modify the code can understand some of the reasoning in case they have another use-case.